src/main/webapp/app/pages/companyManagement/CompanyDetailsPage.tsx

```diff
@@ -0,0 +1,203 @@
+import _ from 'lodash';
+import React from 'react';
+import { match } from 'react-router';
+import { action, observable } from 'mobx';
+import { Col, Row } from 'react-bootstrap';
+import { inject, observer } from 'mobx-react';
+import { RouterStore } from 'mobx-react-router';
+import client from 'app/shared/api/clientInstance';
+import styles from './CompanyDetailsPage.module.scss';
+import { CompanyDTO } from 'app/shared/api/generated/API';
+import OncoKBTable, {
+  SearchColumn,
+} from 'app/components/oncokbTable/OncoKBTable';
+import { getSectionClassName } from 'app/pages/account/AccountUtils';
+import { notifyError } from 'app/shared/utils/NotificationUtils';
+import { filterByKeyword } from 'app/shared/utils/Utils';
+import { PAGE_ROUTE } from 'app/config/constants';
+import LoadingIndicator, {
+  LoaderSize,
+} from '../../components/loadingIndicator/LoadingIndicator';
+import { Link } from 'react-router-dom';
+import { QuickToolButton } from '../userPage/QuickToolButton';
+
+@inject('routing')
+@observer
+export default class CompanyDetailsPage extends React.Component<{
+  routing: RouterStore;
+  match: match;
+}> {
+  @observable companies: CompanyDTO[] = [];
+  @observable areCompaniesLoaded = false;
+
+  constructor(props: Readonly<{ routing: RouterStore; match: match }>) {
+    super(props);
+    this.getCompanies();
+  }
+
+  @action
+  getCompanies() {
```

Comment:
If you want to use the status of loading companies, maybe use `remoteData` instead?